app/validators/blocks/block_validator.py

```diff
@@ -56,16 +59,23 @@ def validate_list_source_reference(self, identifiers):
             if identifier not in self.questionnaire_schema.list_names:
                 self.add_error(self.LIST_REFERENCE_INVALID, id=identifier)
 
-    def validate_answer_source_reference(self, identifiers):
+    def validate_answer_source_reference(self, identifiers, has_selector=False):
+        answers_with_context = self.questionnaire_schema.answers_with_context
+
         for identifier in identifiers:
-            if identifier not in self.questionnaire_schema.answers_with_context:
+            if identifier not in answers_with_context:
                 self.add_error(self.ANSWER_REFERENCE_INVALID, referenced_id=identifier)
-            elif (
-                self.questionnaire_schema.answers_with_context[identifier]["block"]
-                == self.block["id"]
-            ):
+            elif answers_with_context[identifier]["block"] == self.block["id"]:
                 self.add_error(self.ANSWER_SELF_REFERENCE, referenced_id=identifier)
 
+            if (
+                has_selector
+                and answers_with_context[identifier]["answer"]["type"] != "Address"
+            ):
+                self.add_error(
+                    self.SELECTOR_USED_FOR_NON_ADDRESS_TYPE, referenced_id=identifier
+                )
```

Comment:
This enforces that "selector" can only be used for addresses but I think we should update this to something like `if has_selector and identifier not in ANSWER_TYPES_WITH_COMPOSITE_VALUES` where this can just be `ANSWER_TYPES_WITH_COMPOSITE_VALUES = {"Address"}` for now.
Thoughts?